### PR TITLE
LAS/BIN empty point cloud support

### PR DESCRIPTION
**Pull request by WargodHernandez**
Allow loading/saving empty LAS and BIN point clouds #956

Currently doesn't try loading any meta data from the LAS files after the point cloud is determined to be empty. I am not familiar enough with LAS to go any further. 

The missing meta data is:
``` c++
loadedCloud->setMetaData(LAS_SCALE_X_META_DATA, QVariant(lasScale.x));
loadedCloud->setMetaData(LAS_SCALE_Y_META_DATA, QVariant(lasScale.y));
loadedCloud->setMetaData(LAS_SCALE_Z_META_DATA, QVariant(lasScale.z));
loadedCloud->setMetaData(LAS_OFFSET_X_META_DATA, QVariant(lasOffset.x));
loadedCloud->setMetaData(LAS_OFFSET_Y_META_DATA, QVariant(lasOffset.y));
loadedCloud->setMetaData(LAS_OFFSET_Z_META_DATA, QVariant(lasOffset.z));

loadedCloud->setMetaData(LAS_VERSION_MAJOR_META_DATA, QVariant(lasHeader.versionMajor()));
loadedCloud->setMetaData(LAS_VERSION_MINOR_META_DATA, QVariant(lasHeader.versionMinor()));
loadedCloud->setMetaData(LAS_POINT_FORMAT_META_DATA, QVariant(pointFormat));
```

The LAS load also does not attempt any SFs after 0 points are found

### CODE REVIEW

**Comment by WargodHernandez** [comment]
Also Added PCD support

**Comment by WargodHernandez** [comment]
Fixed some 3d mouse related issues

**Comment by dgirardeau** [comment]
I realized that I didn't submit my review of the first commits!

**Comment by dgirardeau** [comment]
And I'll have to test your branch with my 2 3D mice on my side... I fixed last week some stuff in the 3DMouseInput class, but I'm not sure it's still necessary to push that ;)

File: plugins/core/Standard/qPCL/PclIO/PcdFilter.cpp [hunk 1]
```diff
@@ -96,6 +96,7 @@ CC_FILE_ERROR PcdFilter::saveToFile(ccHObject* entity, const QString& filename,
 	PCLCloud::Ptr pclCloud = cc2smReader(ccCloud).getAsSM();
 	if (!pclCloud)
 	{
+		ccLog::Print("!pclCloud");
```
[the remark this answers is not in the document]
Comment (WargodHernandez) [reply plugins/core/Standard/qPCL/PclIO/PcdFilter.cpp]: Yep understood, I just do most debugging in release mode with console messages, I had a lot more in there before the commits :)

File: plugins/core/Standard/qPCL/PclIO/PcdFilter.cpp [hunk 2]
```diff
@@ -47,8 +47,8 @@ PcdFilter::PcdFilter()
                     QStringList{ "pcd" },
                     "pcd",
                     QStringList{ "Point Cloud Library cloud (*.pcd)" },
-                    QStringList(),
-                    Import
+                    QStringList{ "Point Cloud Library cloud (*.pcd)" },
+                    Import | Export
```
[the remark this answers is not in the document]
Comment (WargodHernandez) [reply plugins/core/Standard/qPCL/PclIO/PcdFilter.cpp]: Thanks :)

**Pull request merged**